src/hera/workflows/script.py

```diff
@@ -497,6 +497,15 @@ class will be used as inputs, rather than the class itself.
             else:
                 default = MISSING
 
+            type_ = get_origin(func_param.annotation)
+            args = get_args(func_param.annotation)
+            if type_ is Annotated:
+                type_ = get_origin(args[0])
+                args = get_args(args[0])
+
+            if (type_ is Union and len(args) == 2 and args[1] is type(None)) and default is MISSING:
```

Comment:
Technically this should be `any(arg is type[None] for arg in args` or more simply `type(None in args)` - I wrote a function that ended up unused here https://github.com/argoproj-labs/hera/blob/3470e8e8ec6f416d66e1e9a2a35970f3cbcf1261/src/hera/workflows/_meta_mixins.py#L498-L509

Maybe you could adapt it for use here? Otherwise it might be worth taking the type inspect code from `type-lens` https://github.com/litestar-org/type-lens/blob/670000521ec55e81b2a25ace0e669c38c9ca8e8b/type_lens/type_view.py#L151-L154

Reply:
Good point. I updated this as `type(None) in args`. And I will add additional PR after this to refine these type handling for more unified way.

https://github.com/argoproj-labs/hera/pull/1147#issuecomment-2286706343